cobra/test/test_flux_analysis.py

```diff
@@ -387,25 +387,56 @@ def test_phenotype_phase_plane(self, model):
             pytest.skip("can't test plots without 3D plotting")
         data.plot()
 
-    def check_entries(self, out, desired_entries):
-        """ensure each entry in desired_entries appears in output"""
-        output = out.getvalue().strip()
-        output_set = set((re.sub('\s', '', l) for l in output.splitlines()))
-        for item in desired_entries:
-            assert any(re.sub('\s', '', item) in line for line in output_set)
+    def check_line(self, output, expected_entries, pattern=re.compile(r"\s")):
+        """Ensure each expected entry is in the output."""
+        output_set = set(pattern.sub("", line) for line in output.splitlines())
+        for elem in expected_entries:
+            assert pattern.sub("", elem) in output_set
+
+    def check_in_line(self, output, expected_entries,
+                      pattern=re.compile(r"\s")):
+        """Ensure each expected entry is contained in the output."""
+        output_strip = [pattern.sub("", line) for line in output.splitlines()]
+        for elem in expected_entries:
+            assert any(pattern.sub("", elem) in line for line in output_strip)
 
     @pytest.mark.skipif((pandas is None) or (tabulate is None),
                         reason="summary methods require pandas and tabulate")
-    @pytest.mark.parametrize("solver", optlang_solvers)
-    def test_summary_methods(self, model, solver):
-        model.solver = solver
-        # Test model summary methods
-        with warnings.catch_warnings():
-            warnings.simplefilter("error", UserWarning)
-            with pytest.raises((UserWarning, RuntimeError)):
-                model.summary()
+    def test_model_summary_unoptimized(self, model, opt_solver):
+        model.solver = opt_solver
+        with pytest.raises(RuntimeError):
+            model.summary()
 
-        desired_entries = [
+    @pytest.mark.skipif((pandas is None) or (tabulate is None),
+                        reason="summary methods require pandas and tabulate")
+    def test_model_summary(self, model, opt_solver):
+        model.solver = opt_solver
+        # test non-fva version (these should be fixed for textbook model
+        expected_entries = [
+            'o2_e      21.8',
+            'glc__D_e  10',
+            'nh4_e      4.77',
+            'pi_e       3.21',
+            'h2o_e  29.2',
+            'co2_e  22.8',
+            'h_e    17.5',
+            'Biomass_Ecol...  0.874',
+        ]
+        # Need to use a different method here because
+        # there are multiple entries per line.
+        model.optimize()
+        with captured_output() as (out, err):
+            model.summary()
+        self.check_in_line(out.getvalue(), expected_entries)
+
+    @pytest.mark.skipif((pandas is None) or (tabulate is None),
```

Comment:
True, I just copied the summary test which still had the statements. I guess the test files are not cleaned up in this regard.